exporter/prometheusremotewriteexporter/exporter.go

```diff
@@ -274,11 +295,25 @@ func (prwe *PrwExporter) export(ctx context.Context, tsMap map[string]*prompb.Ti
 		return errs
 	}
 
-	input := make(chan *prompb.WriteRequest, len(requests))
-	for _, request := range requests {
-		input <- request
+	if !prwe.walEnabled() {
+		return prwe.exportWriteRequests(ctx, requests)
 	}
-	close(input)
+
+	// Otherwise, serialize the requests to the WAL.
+	if err := prwe.writeToWAL(requests); err != nil {
+		errs = append(errs, consumererror.Permanent(err))
+	}
+	return errs
+}
+
+func (prwe *PrwExporter) exportWriteRequests(ctx context.Context, requests []*prompb.WriteRequest) (errs []error) {
```

Comment:
It is not a go pattern to return slice of errors.

Reply:
Following precedent with the prior code
<img width="1274" alt="Screen Shot 2021-05-18 at 10 39 53 AM" src="https://user-images.githubusercontent.com/4898263/118611345-6839f080-b7c5-11eb-9e61-0317ab95f6ba.png">


It is also a normal Go idiomatic code to return multiple errors uncombined if you are processing multiple operations. Please see:
* https://pkg.go.dev/cloud.google.com/go/bigtable#Table.ApplyBulk

Later on however, we could use Hashicorp's multierror but for now keeping code as it was before, and just adapting.

Reply:
Sure, we can address later.